tests/conftest.py

```diff
@@ -196,6 +204,16 @@ def pytest_runtest_setup(item: "pytest.Item"):
         return
     if any(True for mark in item.iter_markers(name="slowtest")):
         pytest.skip("Skipping slow tests")
+    if any(True for mark in item.iter_markers(name="migration_test")):
+        try:
+            file_name: str = item.location[0]
+            date: str = re.search("_v(.*)_to_v(.*)[0-9].py", file_name).group(2)
+            test_creation_date: datetime.datetime = datetime.datetime(int(date[0:4]), int(date[4:6]), int(date[6:8]))
```

Comment:
This is slightly more efficient, because the regex doesn't have to search where in the string the pattern starts. Next to that this provides guarantees that the cast to int() will succeed.
```suggestion
            date: str = re.fullmatch("test_v[0-9]{9}_to_v([0-9]{9}).py", file_name).group(1)
            test_creation_date: datetime.datetime = datetime.datetime(int(date[0:4]), int(date[4:6]), int(date[6:8]))
```

Reply:
I needed to change this a bit but the idea was kept :)